### PR TITLE
Remove old files and use new templates [SOLENG-833]

### PR DESCRIPTION
**Pull request by samuelallan72**
Note that check.yaml differs from the snap_check template in automation, because currently the snap_check template does not have a job for unit tests.

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -8,27 +13,35 @@ on:
...
+        run: echo "SNAP_FILE=$(find . -name "*.snap")" >> $GITHUB_ENV
+
       - uses: snapcore/action-publish@v1
         env:
           SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.STORE_LOGIN }}
         with:
-          snap: ${{ steps.build.outputs.snap }}
-          release: latest/edge,3/edge
+          snap: ${{ env.SNAP_FILE }}
+          release: latest/edge
```
Comment (rgildein) [inline .github/workflows/release.yaml:47]: We need to release to both channels.

**Review by rgildein — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (samuelallan72) [reply .github/workflows/release.yaml]: Thanks, addressed :)

**Comment by jneo8** [comment]
Re-trigger functional test

**Review by rgildein — DISMISSED** [review]
LGTM, however, maybe we will need to fix func test for Python 3.10.

**Review by chanchiwai-ray — APPROVED** [review]
A reminder to also update: https://github.com/canonical/solutions-engineering-automation/blob/main/.github/workflows/weekly_tests.yaml#L51

**Comment by samuelallan72** [comment]
Thanks @chanchiwai-ray , added in https://github.com/canonical/solutions-engineering-automation/pull/116 :)

**Review by rgildein — APPROVED** [review]

**Pull request merged**